src/components/Container.js

```diff
@@ -131,58 +120,72 @@ class Container extends React.Component{
   }
 
   indexNamebyElevation = () => {
-    const { selectedTrail } = this.state;
-    let newHydrants = _.cloneDeep(this.state.hydrants);
+    const { selectedTrail, hydrants } = this.state;
 
-    const sortedTrailHydrants = _.chain(newHydrants)
-      .pickBy((h) => h['trail'] === selectedTrail)
+    const sortedTrailHydrants = hydrants.toJS()
+      .filter((h) => h.trail === selectedTrail)
       .orderBy('elevation', 'desc')
       .map((h,i)=> {
         h.name = i + 1
         return h
       }).value();
 
-    _.each(sortedTrailHydrants , (h,key)=> {
-      newHydrants[h.id] = h
+    let newHydrants = hydrants.map((h) => {
+      const elevationIndex = _.findIndex(sortedTrailHydrants, (sortedHydrant) => sortedHydrant.id === h.get('id'));
+      return h.set('name', elevationIndex);
     });
+
     this.setState({
       hydrants: newHydrants
     });
   }
 
-  mapControlClicked = (type) => {
-    if (type === 'Hydrant') {
-      this.setState({createType: type});
-    }else {
-      this.setState({createType: type || null, selectedTrail: null});
+  importKMLClicked = (klmData) => {
+    console.log(klmData)
+    if( this.state.mode === 'trails'){
+      this.setState({
+        trails: Immutable.fromJS(klmData)
+      })
+    } else {
+      this.setState({
+        hydrants: Immutable.fromJS(klmData)
+      })
     }
   }
 
+
   render(){
-    const {trails, createType, selectedTrail, hydrants} = this.state;
+    const {trails, mode, selectedTrail, hydrants} = this.state;
+
+    console.log(this.state.trails.toJS())
 
     return (
       <div style={{position: 'relative'}}>
         <OpenLayersMap
-          createType={createType}
+          mode={mode}
           endDraw={this.endDraw}
           endModify={this.endModify}
           trails={trails}
           hydrants={hydrants}
           selectedTrail={selectedTrail}
         />
         <MapControls
-          onClick={this.mapControlClicked}
-          canAddHydrant={!!selectedTrail}
+          mode={mode}
+          changeMode={(mode) => this.setState({mode})}
         />
         <TrailList
-          renameTrail={this.renameTrail}
+          modifyTrail={this.modifyTrail}
           trails={trails}
           hydrants={hydrants}
           selected={selectedTrail}
-          deleteTrail={this.deleteTrail}
-          trailSelected={(id) => this.setState({selectedTrail: id, createType: id ? 'Trail' : null})}
+          trailSelected={(id) => this.setState({selectedTrail: id})}
         />
+
+        <ImportExport
+        mode = {mode}
```

Comment:
Nitpick: indent the props. Did you get your linter setup? It should catch this.

Reply:
Indented. I have the linter setup but it doesn't like this:

```js
  renderTrailsDebounce = _.debounce(this.renderTrails, 50);
```
it throws a parsing error at the "=" and gives up linting.